add sandcastle device type test base discovery

walterddr opened the pull request:
Differential Revision: D24649868

mruberry commented on torch/testing/_internal/common_device_type.py:
```diff
@@ -20,6 +20,10 @@
 except ImportError:
     HAS_PSUTIL = False
 
+IS_SANDCASTLE = os.getenv('SANDCASTLE') == '1' or os.getenv('TW_JOB_USER') == 'sandcastle'
```
I think you can import at least two of these, and the third one should probably be added to and imported from the same place. See: 

https://github.com/pytorch/pytorch/blob/789e9353044eb2227462ef470e4b362ac025be4d/torch/testing/_internal/common_utils.py#L61

walterddr replied:
sounds good. will move those to `common_utils`

mruberry reviewed (approved):
Looks great, but I think we can import bools like `IS_SANDCASTLE` from common_utils.py.

Commit:
add sandcastle device type test base discovery (#47119)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/47119

Test Plan:
tests
1. test_cuda still works:
`buck test --no-cache -c test.external_runner=tpx mode/dev-nosan //caffe2/test:cuda -- --use-remote-execution --force-tpx`
2. test_torch is blocked on D24623962
`buck test --no-cache -c test.external_runner=tpx mode/dev-nosan //caffe2/test:torch -- --use-remote-execution --force-tpx`

Reviewed By: mruberry

Differential Revision: D24649868

fbshipit-source-id: 40e89383528e3ce1b21c596fa008eea81135c5f0
